### PR TITLE
Use xml_find_chr where appropriate

### LINKED ISSUE #1151: Use `xml_find_chr()` where applicable
`xml2::xml_text(xml2::xml_find_first(xml, "/x/path"))` is equivalent to `xml2::xml_find_chr(xml, "string(/x/path)")`, looking a bit cleaner.
We could refactor our codebase to use the more concise idiom.

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #1151 

Also introduce a new helper for the very common use case of `xml_find_char(expr, "string(SYMBOL_FUNCTION_CALL)")` (possibly nested at `expr/` or `expr/expr/` etc)

FWIW a quick test suggests it's about 20% faster to use `xml_find_char()` as well.

### CODE REVIEW

File: R/xp_utils.R [hunk 1]
```diff
@@ -113,3 +113,25 @@ xp_and <- function(...) paren_wrap(..., sep = "and")
...
+xp_call_name <- function(expr, depth = 1L, condition = NULL) {
+  if (is.null(condition)) {
+    node <- "SYMBOL_FUNCTION_CALL"
+  } else {
+    node <- sprintf("SYMBOL_FUNCTION_CALL[%s]", condition)
+  }
+
+  # use an explicit condition for the most common cases, and an uglier
+  #   expression for full generality
+  if (depth == 0L) {
```
Comment (AshesITR) [inline R/xp_utils.R:126]: Should use `strrep()` here.

Comment (MichaelChirico) [reply R/xp_utils.R]: good idea! much cleaner

File: R/paste_linter.R [hunk 2]
```diff
@@ -56,10 +53,7 @@ paste_linter <- function(allow_empty_sep = FALSE, allow_to_string = FALSE) {
         and SYMBOL_SUB[text() = 'collapse']/following-sibling::expr[1][STR_CONST]
       ]")
       to_string_expr <- xml2::xml_find_all(xml, to_string_xpath)
-      collapse_value <- get_r_string(xml2::xml_find_first(
-        to_string_expr,
-        "./SYMBOL_SUB[text() = 'collapse']/following-sibling::expr[1]"
-      ))
+      collapse_value <- get_r_string(to_string_expr, xpath = "./SYMBOL_SUB[text() = 'collapse']/following-sibling::expr[1]")
```
Comment (AshesITR) [inline R/paste_linter.R:56]: line_length

**Comment by AshesITR** [comment]
Would be a good time to delint R/utils.R (implicit_integer)

**Comment by AshesITR** [comment]
LGTM, just a few lints.

**Comment by AshesITR** [comment]
For some reason, line 86 didn't produce a check warning:
```
defaults <- vapply(x[missing], default_name, character(1), USE.NAMES = FALSE)
```

(R/utils.R:89)

**Review by AshesITR — APPROVED** [review]

**Pull request merged**